captum/attr/_core/layer/layer_integrated_gradients.py

```diff
@@ -294,6 +312,13 @@ def layer_forward_hook(module, hook_inputs, hook_outputs=None):
                 grads = torch.autograd.grad(torch.unbind(output), inputs)
             return grads
 
+        def extract_device(module, hook_inputs, hook_outputs):
+            if hook_inputs is not None and len(hook_inputs) > 0:
+                return hook_inputs[0].device
+            if hook_outputs is not None and len(hook_outputs) > 0:
+                return hook_outputs[0].device
+            return next(module.parameters()).device
```

Comment:
I moved the check into this function (haven't pushed the change yet). So, if we have the check on top of this function we don't need to check the length of `module.parameters()`, do we ? Because otherwise the check will fail ?
`
(hook_inputs is None or len(hook_inputs) == 0)
                and (hook_outputs is None or len(hook_outputs) == 0)
                and len(list((module.parameters()))) == 0
`

Reply:
Actually `len(list((module.parameters())))` is going to exhaust all elements it is not right way of  checking the length - let me fix that

Reply:
Makes sense, yeah if you move the check you don't need to check length again! But I guess inputs and output checks are already there, so might be easier to just add the last check to the end, and just raise an exception if none of the return conditions hit as opposed to putting the full check above? But either way works!

Reply:
Also doesn't each separate call to module.parameters() return a new iterator?

Reply:
1. For `module.parameters() `, it looks like it is creating a new iterator every time. Good point. But it's better to create it ones in our case.
2. Yeah, I could have the last check on the parameters later but in that case, I'll need to throw new error with a new message and it is probably better to combine the checks in one place.